src/ferui-components/tree-view/tree-node-component.ts

```diff
@@ -176,9 +181,14 @@ export class FuiTreeNodeComponent<T> implements OnInit, OnDestroy, DoCheck {
 
   /**
    * Calculates the needed padding based on nodes level and if it has children
+   * 20 for the indentation pixels per level
+   * 10 pixels if has children to add equal 5 pixel margin around icon
+   * 30 pixels if no children to add margin from beginning of node text to beginning of left hand tree view
    */
   calculatePadding(): number {
```

Comment:
Replace this function by a simple variable and re-calculate the padding value only when you need to (when the user is resizing the window for instance or something similar).

Reply:
Okay fixed! This is now only being used when we need to actually calculate the node's padding, not on the template anymore

Reply:
If it is not used by the template, you can make it private then ;-).

Reply:
Done!